examples/jsm/exporters/GLTFExporter.js

```diff
@@ -839,7 +860,7 @@ class GLTFWriter {
 
 		let componentSize;
 
-		if ( componentType === WEBGL_CONSTANTS.UNSIGNED_BYTE ) {
+		if ( componentType === WEBGL_CONSTANTS.UNSIGNED_BYTE || componentType === WEBGL_CONSTANTS.BYTE ) {
```

Comment:
Not opposed to having this, but did you mean to include these lines?

Reply:
Ah, I was trying to (and partially succeeding) exporting compressed meshes as well, but that could / should be a separate PR. Some of that is described in my original post.

Reply:
Ok, I see! 

> ...my understanding is that some of them would need to be converted back to non-meshopt formats (e.g. byte normalized back to float for vertex positions)

It looks like you're part way to implementing https://github.com/mrdoob/three.js/issues/20474. No need to convert anything back to float32, I think we just need to detect when we're writing accessors in a format that the core specification doesn't allow, and append `KHR_mesh_quantization` to `extensionsUsed` and `extensionsRequired`. Loading a meshopt-compressed model with GLTFLoader will not leave any meshopt-specific data in the resulting scene, but does leave data that depends on quantization.

Up to you whether you'd prefer to keep the changes in this PR, I'm okay with it either way!